src/mz/src/secrets.rs

```diff
@@ -0,0 +1,231 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::{fmt::Display, fs::read_to_string};
+
+use anyhow::{bail, Context, Result};
+use clap::{Parser, Subcommand};
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    configuration::ValidProfile,
+    region::{get_provider_region_environment, CloudProviderRegion},
+};
+
+#[derive(Debug, Subcommand)]
+#[clap(group = clap::ArgGroup::new("input").multiple(false))]
+pub enum SecretCommand {
+    /// Alter a secret
+    Alter {
```

Comment:
I propose we call this "update", since we're using the standard CLI verbs here rather than the SQL verbs. (E.g., "delete" not "drop.") 

```suggestion
    /// Update a secret
    Update {
```